### PR TITLE
cardano-testnet: fix that genesis files generated by create-testnet-data were ignored

### PR DESCRIPTION
**Pull request by smelc**
# Description

* Fixes that genesis files generated by `create-testnet-data` were not passed to the nodes spawned in clusters
* As a consequence, become able to benefit of `create-testnet-data` new features of creating DReps; which is used in this PR to remove custom inialization

:warning: This PR changes the shape of directories generates by `cardano-testnet`. It changes from:

```shell
.
├── alonzo-genesis.json
├── byron
│   └── genesis.json
├── byron.genesis.spec.json
├── configuration.yaml
├── conway-genesis.json
├── genesis.alonzo.spec.json
├── genesis.conway.spec.json
└── shelley
    ├── genesis.alonzo.json
    ├── genesis.conway.json
    └── genesis.shelley.json
```

to

```shell
.
├── alonzo-genesis.json
├── byron
│   └── genesis.json
├── byron.genesis.spec.json
├── configuration.yaml
├── conway-genesis.json
├── genesis.alonzo.spec.json
├── genesis.conway.spec.json
└── shelley-genesis.json
```

# Checklist

- [X] Commit sequence broadly makes sense and commits have useful messages
- [X] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

### CODE REVIEW

File: cardano-testnet/src/Testnet/Components/Configuration.hs [hunk 1]
```diff
@@ -149,27 +145,26 @@ createSPOGenesisAndFiles (NumPools numPoolNodes) era shelleyGenesis alonzoGenesi
...
-  H.renameFile (tempAbsPath </> "shelley-genesis.json") (genesisShelleyDir </> "genesis.shelley.json")
 
   -- For some reason when setting "--total-supply 10E16" in create-testnet-data, we're getting negative
-  -- treasury. TODO: This should be fixed by https://github.com/IntersectMBO/cardano-cli/pull/644
-  -- So this can be removed when cardano-cli is upgraded above 8.20.3.0.
-  H.rewriteJsonFile @Value (genesisShelleyDir </> "genesis.shelley.json") $ \o -> o
+  -- treasury. TODO: This was supposed to be fixed by https://github.com/IntersectMBO/cardano-cli/pull/644,
+  -- but no, it's still there.
+  H.rewriteJsonFile @Value (tempAbsPath </> defaultGenesisFilepath ShelleyEra) $ \o -> o
     & L.key "maxLovelaceSupply" . L._Integer .~ 10_000_000_000_000_000
```
Comment (smelc) [inline cardano-testnet/src/Testnet/Components/Configuration.hs:163]: I tried to get rid of this rewrite, but it makes the `Treasury Growth` fail so something is still wrong somewhere. I will look into it in another PR. I don't want to block this one.

File: cardano-testnet/src/Testnet/Components/Configuration.hs [hunk 2]
```diff
@@ -56,12 +57,12 @@ import qualified Hedgehog.Extras.Test.Base as H
 import qualified Hedgehog.Extras.Test.File as H
 
 -- | Returns JSON encoded hashes of the era, as well as the hard fork configuration toggle.
-createConfigYaml :: ()
+createConfigJSON :: ()
```
Comment (carbolymer) [inline cardano-testnet/src/Testnet/Components/Configuration.hs:60]: Small nitpick, [acronyms should be camelcased this way](https://github.com/input-output-hk/cardano-node-wiki/wiki/Style-guide#naming-convention):
```suggestion
createConfigJson :: ()
```

**Review by carbolymer — APPROVED** [review]

File: cardano-testnet/src/Testnet/Components/Configuration.hs (hunk 2 above)
Comment (smelc) [reply cardano-testnet/src/Testnet/Components/Configuration.hs:60]: Thanks for the pointer @carbolymer :+1: Changing.

File: cardano-testnet/src/Testnet/Components/Configuration.hs [hunk 3]
```diff
@@ -95,30 +97,29 @@ createSPOGenesisAndFiles
...
-  let genesisShelleyFpAbs = tempAbsPath </> defaultGenesisFilepath ShelleyEra
-      genesisAlonzoFpAbs = tempAbsPath </> defaultGenesisFilepath AlonzoEra
-      genesisConwayFpAbs = tempAbsPath </> defaultGenesisFilepath ConwayEra
-      genesisShelleyDirAbs = takeDirectory genesisShelleyFpAbs
+createSPOGenesisAndFiles (NumPools numPoolNodes) era shelleyGenesis alonzoGenesis conwayGenesis (TmpAbsolutePath tempAbsPath) = GHC.withFrozenCallStack $ do
+  let inputGenesisShelleyFp = tempAbsPath </> genesisInputFilepath ShelleyEra
+      inputGenesisAlonzoFp  = tempAbsPath </> genesisInputFilepath AlonzoEra
+      inputGenesisConwayFp  = tempAbsPath </> genesisInputFilepath ConwayEra
+
+  -- We write the genesis files to disk, to pass them to create-testnet-data.
```
Comment (Jimbo4350) [inline cardano-testnet/src/Testnet/Components/Configuration.hs:104]: These files shouldn't be modified after they are created. `create-testnet-data` should create the final genesis files we want to use and then we create a configuration file based on those genesis files.

File: cardano-testnet/src/Testnet/Defaults.hs [hunk 4]
```diff
@@ -552,7 +551,9 @@ defaultMainnetTopology =
   in RealNodeTopology [single]
 
 defaultGenesisFilepath :: CardanoEra a -> FilePath
-defaultGenesisFilepath era = "shelley" </> ("genesis." <> eraToString era <> ".json")
+defaultGenesisFilepath era =
+  -- This path is actually generated by create-testnet-data. Don't change it.
+  eraToString era <> "-genesis.json"
```
Comment (Jimbo4350) [inline cardano-testnet/src/Testnet/Defaults.hs:559]: :+1

File: cardano-testnet/src/Testnet/Defaults.hs [hunk 5]
```diff
@@ -555,6 +558,18 @@ defaultGenesisFilepath era =
   -- This path is actually generated by create-testnet-data. Don't change it.
   eraToString era <> "-genesis.json"
 
+-- | The relative path to DRep keys in directories created by cardano-testnet
+defaultDRepVkeyFp
+  :: Int -- ^ The DRep's index (starts at 0)
+  -> FilePath
+defaultDRepVkeyFp n = "drep-keys" </> ("drep" <> show n) </> "drep.vkey"
```
Comment (Jimbo4350) [inline cardano-testnet/src/Testnet/Defaults.hs:565]: :+1:

File: cardano-testnet/src/Testnet/Components/Configuration.hs [hunk 6]
```diff
@@ -56,14 +56,13 @@ import qualified Hedgehog.Extras.Stock.Time as DTC
...
--- TODO rename me, it generates JSON
 -- | Returns JSON encoded hashes of the era, as well as the hard fork configuration toggle.
-createConfigYaml :: ()
+createConfigJson :: ()
   => (MonadTest m, MonadIO m, HasCallStack)
   => TmpAbsolutePath
   -> AnyCardanoEra -- ^ The era used for generating the hard fork configuration toggle
   -> m LBS.ByteString
-createConfigYaml (TmpAbsolutePath tempAbsPath) era = GHC.withFrozenCallStack $ do
+createConfigJson (TmpAbsolutePath tempAbsPath) era = GHC.withFrozenCallStack $ do
```
Comment (Jimbo4350) [inline cardano-testnet/src/Testnet/Components/Configuration.hs:65]: Let's use pretty JSON here. Makes debugging a little easier.

File: cardano-testnet/src/Testnet/Components/Configuration.hs [hunk 7]
```diff
@@ -115,14 +115,10 @@ createSPOGenesisAndFiles (NumPools numPoolNodes) era shelleyGenesis alonzoGenesi
...
       startTime = sgSystemStart shelleyGenesis
 
-  -- TODO: Remove this rewrite.
+ -- TODO: Remove this rewrite.
  -- 50 second epochs
  -- Epoch length should be "10 * k / f" where "k = securityParam, f = activeSlotsCoeff"
   H.rewriteJsonFile @Value inputGenesisShelleyFp $ \o -> o
-    -- TODO: remove rho and tau adjustment after https://github.com/IntersectMBO/cardano-api/pull/425 gets
-    -- integrated with newer cardano-api into node
-    & L.key "protocolParams" .  L.key "rho" . L._Number  .~ 0.1
```
Comment (Jimbo4350) [inline cardano-testnet/src/Testnet/Components/Configuration.hs:124]: :muscle:

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Conway/DRepRetirement.hs [hunk 8]
```diff
@@ -93,49 +93,6 @@ hprop_drep_retirement = H.integrationRetryWorkspace 2 "drep-retirement" $ \tempA
       drepSKeyFp :: Int -> FilePath
       drepSKeyFp n = tempAbsPath' </> "drep-keys" </> ("drep" <> show n) </> "drep.skey"
 
-  -- Create Drep registration certificates
```
Comment (Jimbo4350) [inline cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Conway/DRepRetirement.hs:96]: Do we still test this work flow?

**Review by Jimbo4350 — CHANGES_REQUESTED** [review]
Nice work :muscle:

**Review by Jimbo4350 — APPROVED** [review]
LGTM! Only on concern: https://github.com/IntersectMBO/cardano-node/pull/5732#discussion_r1547952322

File: cardano-testnet/src/Testnet/Components/Configuration.hs (hunk 3 above)
Comment (smelc) [reply cardano-testnet/src/Testnet/Components/Configuration.hs:104]: Yep agreed, we should remove the last rewrites that remain later on :+1:

File: cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Conway/DRepRetirement.hs (hunk 8 above)
Comment (smelc) [reply cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Conway/DRepRetirement.hs:96]: No we don't, but I'll readd such a test shortly afterwards.

Follow-up tracked here: https://github.com/IntersectMBO/cardano-node/issues/5742

File: cardano-testnet/src/Testnet/Components/Configuration.hs (hunk 6 above)
Comment (smelc) [reply cardano-testnet/src/Testnet/Components/Configuration.hs:65]: Good point, I'm myself always reformatting those JSON files before inspecting them :laughing:

File: cardano-testnet/src/Testnet/Components/Configuration.hs (hunk 3 above)
Comment (Jimbo4350) [reply cardano-testnet/src/Testnet/Components/Configuration.hs:105]: Can you update the comment so it's obvious this isn't the behaviour we want?

**Pull request merged**